core/src/main/kotlin/com/strumenta/kolasu/traversing/Structurally.kt

```diff
@@ -159,15 +159,12 @@ fun <T> Node.collectByType(klass: Class<T>, walker: KFunction1<Node, Sequence<No
 
 fun Node.freeze() {
     walkLeavesFirst().forEach {
-        if(!Node.Freezer.childrenMap.containsKey(it)) {
-            Node.Freezer.childrenMap.put(it, it.children)
-        }
+        Node.childrenMap.putIfAbsent(it, it.children)
```

Comment:
This will still compute `it.children` every time though. I think there's an overload taking a closure. Anyway, probably `children` will be called only once so I don't think it will be a problem most of the time.